### PR TITLE
MDCT-2041: Page-level status indicators

### PR DESCRIPTION
**Pull request by ntsummers1**
## Description
Fixes [MDCT-2041](https://qmacbis.atlassian.net/browse/MDCT-2041)

Adds page level status indicators if you've completed a plan on a DrawerReport page!

https://user-images.githubusercontent.com/19439679/199075358-1e6f1568-7207-4d07-a456-ebc1e89ad6fc.mov

### How to test

1. Create a program
2. Go to Section A: Add Plans
3. Create a plan or as many plans as you like
4. Navigate to Section D: Program Characteristics
5. Complete one of the plans you've created by hitting enter on the plan and filling out the drawer
6. Note that the plan you entered info for and saved and closed now has a green checkmark and any other plans you created dont
7. Nagivate to Section D: Financial Performance (Or any other section in D other then Quality Measures and Sanctions)
8. Repeat 5-6 as much as you like!

## Code author checklist
- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [x] I have added analytics, if necessary
- [x] I have updated the documentation, if necessary

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [x] I have done the lgtm review

### CODE REVIEW

File: services/ui-src/src/components/reports/DrawerReportPage.test.tsx [hunk 1]
```diff
@@ -35,6 +35,24 @@ const mockReportContextWithoutEntities = {
...
+  ...mockReportContext,
+  report: {
+    ...mockReportContext.report,
+    fieldData: {
+      ...mockReportContext.report.fieldData,
+      plans: [
+        {
+          id: 123,
+          name: "example-plan1",
+          "mock-drawer-text-field": "example-explaination",
```
Comment (braxex) [inline services/ui-src/src/components/reports/DrawerReportPage.test.tsx:48]: Typo: 
```suggestion
          "mock-drawer-text-field": "example-explanation",
```

File: services/ui-src/src/components/reports/DrawerReportPage.tsx [hunk 2]
```diff
@@ -96,21 +98,39 @@ export const DrawerReportPage = ({ route }: Props) => {
...
-    ));
+  const entityRows = (entities: EntityShape[]) => {
+    return entities.map((entity) => {
+      /*
+       * If the entity has any data from any drawerForms fields,
+       * it must have been completed at some point. So simply check to
+       * see if the first drawerForm field has been entered and saved
+       * to the entity
+       */
+      const isEntityCompleted = drawerForm.fields?.[0]?.id in entity;
```
Comment (braxex) [inline services/ui-src/src/components/reports/DrawerReportPage.tsx:109]: Suggestion: We could probably future-proof this and make it so we don't need the comment if we just check for each of the fields in the entity with something like this:

```suggestion
      const isEntityCompleted = drawerForm.fields?.forEach((field: whatever) => field.id in entity);
```

That way we're explicitly checking for full completion. Just a thought.

**Review by braxex — APPROVED** [review]
A couple notes but nothing blocking. Nice work!

**Review by scwambach — APPROVED** [review]

File: services/ui-src/src/components/reports/DrawerReportPage.tsx (hunk 2 above)
Comment (ntsummers1) [reply services/ui-src/src/components/reports/DrawerReportPage.tsx]: Paired it down to:

`const isEntityCompleted = drawerForm.fields?.every((field: FormField) => field.id in entity);`

I think that gets at what you were looking for

**Review by gmrabian — APPROVED** [review]

**Pull request merged**